app/code/community/OnePica/AvaTax/Model/Observer.php

```diff
@@ -405,6 +417,25 @@ public function salesModelServiceQuoteSubmitBefore(Varien_Event_Observer $observ
     }
 
     /**
+     * Stop order creation if tax estimation has problems
+     *
+     * @param Varien_Event_Observer $observer
+     * @return $this
+     * @throws OnePica_AvaTax_Model_Exception
+     */
+    public function controllerActionPostdispatchCheckoutOnepageSaveShippingMethod(Varien_Event_Observer $observer) {
+        /** @var Mage_Sales_Model_Quote $quote */
+        $quote = Mage::getSingleton('checkout/cart')->getQuote();
```

Comment:
Update _getQuote method to get quote with singleton and use $this->_getQuote instead